### PR TITLE
Add usage timeframe and percentage tiers

### PR DESCRIPTION
**Pull request by faiberrec**

### CODE REVIEW

File: lib/recurly.rb [hunk 1]
```diff
@@ -1,50 +1,7 @@
...
-  require 'recurly/gift_card'
-  require 'recurly/purchase'
-  require 'recurly/webhook'
-  require 'recurly/verify'
-  require 'recurly/tier'
-  require 'recurly/dunning_campaign'
-  require 'recurly/dunning_cycle'
-  require 'recurly/invoice_template'
+
+  Dir.glob('lib/recurly/*.rb').each{ |fn| require fn }
```
Comment (faiberrec) [inline lib/recurly.rb]: I'm highly concerned about this line, will this work?

File: lib/recurly/resource.rb [hunk 2]
```diff
@@ -1118,5 +1118,10 @@ def fetch_associated(name, value, options = {})
     def xml_keys
       changed_attributes.keys.sort
     end
+
+    def add_money_tag?(key, value)
+      value && key.end_with?('_in_cents') && !respond_to?(:currency) && !value.is_a?(Money) &&
+        (!is_a?(PercentageTier) && !is_a?(SubAddOnPercentageTier))
```
Comment (seckenrode) [inline lib/recurly/resource.rb]: **Minor nit** I tend to prefer splitting all of the different segments onto their own lines.  Just marginally more readable IMO.  Very optional.

```suggestion
      value && 
        key.end_with?('_in_cents') && 
        !respond_to?(:currency) && 
        !value.is_a?(Money) &&
        !is_a?(PercentageTier) && 
        !is_a?(SubAddOnPercentageTier)
```

File: lib/recurly/sub_add_on_percentage_tier.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+      usage_percentage
+    )
+
+    def xml_keys
+      attributes.keys
+    end
+
+    embedded! true
+  end
+end
```
Comment (seckenrode) [inline lib/recurly/sub_add_on_percentage_tier.rb]: **Formatting** Ensure we have a new line at the end of the file please.

File: lib/recurly/currency_percentage_tier.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+      currency
+    )
+
+    def xml_keys
+      attributes.keys
+    end
+
+    embedded! true
+  end
+end
```
Comment (seckenrode) [inline lib/recurly/currency_percentage_tier.rb]: **Formatting** ensure newlines at hte end fo files please.

We should get rubocop working in here.

**Review by seckenrode — APPROVED** [review]

**Review by seckenrode — APPROVED** [review]

**Pull request merged**